pipeline/pkg/data/enemy/enemy.go

```diff
@@ -0,0 +1,172 @@
+package enemy
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/genshinsim/gcsim/pipeline/pkg/data/dm"
+	"github.com/genshinsim/gcsim/pipeline/pkg/data/textmap"
+	"github.com/genshinsim/gcsim/pkg/model"
+	"go.uber.org/multierr"
+)
+
+type DataSource struct {
+	monsterExcel         []dm.MonsterExcel
+	monsterDescribeExcel []dm.MonsterDescribeExcel
+	monsterCurveExcel    []dm.MonsterCurveExcel
+	textMap              *textmap.DataSource
+}
+
+var keyRegex = regexp.MustCompile(`\W+`) // for removing spaces
+
+func NewDataSource(root string) (*DataSource, error) {
+	var err error
+	e := &DataSource{}
+	e.monsterExcel, err = loadMonsterExcel(root + "/" + MonsterExcelConfigData)
+	if err != nil {
+		return nil, err
+	}
+	e.monsterDescribeExcel, err = loadMonsterDescribeExcel(root + "/" + MonsterDescribeExcelConfigData)
+	if err != nil {
+		return nil, err
+	}
+	e.monsterCurveExcel, err = loadMonsterCurveExcel(root + "/" + MonsterCurveExcelConfigData)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: crutch to get enemy names
+	e.textMap, err = textmap.NewTextMapSource(root + "/../TextMap/TextMapEN.json")
+	if err != nil {
+		return nil, err
+	}
+	return e, nil
```

Comment:
is there a better way to get the EN textmap?

Reply:
Maybe something like here? https://github.com/genshinsim/gcsim/blob/8d62c1ec28bf6189fed4f1740fe7a90d3055ec6e/pipeline/cmd/generate/main.go#L120

Reply:
fixed a little, but the question is still open
maybe add a general textmap (en) to `cmd/generate/main.go` for cases like this?

Reply:
idk up to you